services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/model/ConditionCheck.java

```diff
@@ -96,24 +118,52 @@ public int hashCode() {
         return result;
     }
 
+    /**
+     * A builder that is used to create a condition check with the desired parameters.
+     * <p>
+     * A valid builder must define both a {@link Key} and an {@link Expression}.
+     */
     public static final class Builder  {
         private Key key;
         private Expression conditionExpression;
 
         private Builder() {
         }
 
+        /**
+         * Sets the primary {@link Key} that will be used together with the condition expression.
+         *
+         * @param key the primary key to use in the operation.
+         * @return a builder of this type
+         */
         public Builder key(Key key) {
             this.key = key;
             return this;
         }
 
+        /**
+         * Sets the primary {@link Key} that will be used together with the condition expression
+         * on the builder by accepting a consumer of {@link Key.Builder}.
+         *
+         * @param keyConsumer a {@link Consumer} of {@link Key}
+         * @return a builder of this type
+         */
         public Builder key(Consumer<Key.Builder> keyConsumer) {
             Key.Builder builder = Key.builder();
             keyConsumer.accept(builder);
             return key(builder.build());
         }
 
+        /**
+         * Defines a logical expression on the table items' key values which, if evaluating to true,
```

Comment:
I think this wording isn't quite correct or is misleading. When I read it, I take it to mean the expression will only be applied to the keys of the record being evaluated. Unless I'm mis-rememebering the expression can act on any attributes and their values in the record, not just limited to the keys.

Reply:
You are right, it's incorrect. I read the documentation again and agree. Re-wording.